src/ui/public/saved_objects/saved_objects_client.js

```diff
@@ -9,20 +9,91 @@ const join = (...uriComponents) => (
   uriComponents.filter(Boolean).map(encodeURIComponent).join('/')
 );
 
+/**
+ * Interval that requests are batched for
+ * @type {integer}
+ */
+const BATCH_INTERVAL = 100;
+
 export class SavedObjectsClient {
   constructor($http, basePath = chrome.getBasePath(), PromiseCtor = Promise) {
     this._$http = $http;
     this._apiBaseUrl = `${basePath}/api/saved_objects/`;
     this._PromiseCtor = PromiseCtor;
+    this.batchQueue = [];
+  }
+
+  /**
+  * Persists an object
+  *
+  * @param {string} type
+  * @param {object} body - { attributes: {}, id: myId }
+  * @param {object} options
+  * @param {boolean} options.overwrite - defaults to false
+  * @returns {promise}
+  */
+  create(type, body = {}, options = {}) {
+    if (!type || !body.attributes) {
+      return this._PromiseCtor.reject(new Error('requires type and attributes'));
+    }
+
+    const url = this._getUrl([type], options);
+
+    return this._request('POST', url, body);
+  }
+
+  /**
+   * Deletes an object
+   *
+   * @param {string} type
+   * @param {string} id
+   * @returns {promise}
+   */
+  delete(type, id) {
+    if (!type || !id) {
+      return this._PromiseCtor.reject(new Error('requires type and id'));
+    }
+
+    return this._request('DELETE', this._getUrl([type, id]));
+  }
+
+  /**
+   * Search for objects
+   *
+   * @param {object} options
+   * @param {string} options.type
+   * @param {string} options.search
+   * @param {string} options.searchFields - see Elasticsearch Simple Query String
+   *                                        Query field argument for more information
+   * @param {integer} options.page - defaults to 1
+   * @param {integer} options.perPage - defaults to 20
+   * @param {array} option.fields - fields to be returned. Returns all unless defined
+   * @returns {promise}
```

Comment:
Can you document what the promises will resolve or reject to for the functions in this file? I'm having a hard time figuring out when `_type`, `_id` `_version` and `_attributes` are used vs `type`, `id`, `version` and `attributes`.     It looks like sometimes its either all `_` or no underscore, but above there is a spot above that handles the response type from this call like this:

```
+              return {
 +                _id: resp.id,
 +                _type: resp.type,
 +                _source: _.cloneDeep(resp._attributes),
 +                found: resp._version ? true : false
 +              };
```

And I'm wondering why id and type don't have underscores but version and attributes too.  Having the comments would help me track that down easier (if only we had interfaces and typescript!)